File: src/Belajar_SOLID_02_OOP_Inheritance/Java/Single_Inheritance.java
package Belajar_SOLID_02_OOP_Inheritance.Java;

class Animal{
    public void walk(){
        System.out.println(getClass().getSimpleName() + " Walk!.. ");
    }
}

class Cat extends Animal{
    @Override
    public void walk() {
        super.walk();
        System.out.println(getClass().getSimpleName() + " Walked!.....");
    }
}

public class Single_Inheritance {
    public static void main(String[] args) {
        Cat kucing = new Cat();
        kucing.walk();
    }
}

File: src/Belajar_SOLID_02_OOP_Inheritance/Kotlin/Single_Inheritance.kt
package Belajar_SOLID_02_OOP_Inheritance.Kotlin

open class Animal{
    open fun walk(){
        println("${javaClass.simpleName} walk!")
    }
}


class Cat:Animal(){
    override fun walk() {
        super.walk()
        println("Yeey ${javaClass.simpleName} walked! ")
    }
}

fun main(){
    val kucing = Cat();
    kucing.walk()
}